scenarios/post_action_regex_openshift_pod_kill.yml

```diff
@@ -0,0 +1,18 @@
+config:
+  runStrategy:
+    runs: 1
+    maxSecondsBetweenRuns: 30
+    minSecondsBetweenRuns: 1
+scenarios:
+  - name: kill up to 3 pods in any openshift namespace
+    steps:
+    - podAction:
+        matches:
+          - namespace: "openshift-.*"
+        filters:
+          - property:
+             name: "state"
+             value: "Running"
+        actions:
+          - checkPodCount:
+              count: 146
```

Comment:
This is going to be a moving target and depends on what optional or configuration dependent operators and associated pods are present.  Example openshift-logging will add n+4 pods where n is the number of nodes.   The real comparison should be before/after counts.  Not sure if powerfulseal can do that, maybe a custom script?   We might have to think about how pre/post checks work together.

Reply:
I do not think powerfulseal can do a changing counts based on a before count. I think they have to be hard coded in the yaml when it gets executed

Here are a couple of solutions I could think of: 

- I could overwrite the post action yaml counts using 2 different possible paths

1. I could add another item in the double array in the config for a pre action yaml or custom script that the response could help fill in the post action yaml counts
2. I could also parse the matches and filters section of the scenario yaml and run a more general kubectl/oc command to get the expected counts to fill in the yaml

- In a lot of the scenarios I could parse the output from powerfulseal running a certain scenario because it spits out how many pods it had to pick from. For the regex scenario I have it picking 3 pods so it doesn't print out the number of pods it had.
I could change the scenario to run 3 times based on the top configuration so I could parse it like the other scenarios and maybe get the number of pods it had to pick from in that output.

- Or like Mike said I could do a custom script with adding options for some of the most common deployed operators and how many pods they create. Here's what I'm thinking. 

1. Set the initial amount of pods in the "openshift-*" namespace (This would be hard coded to whatever number we find, maybe 146?) 
2. Check if certain operators are deployed (find number of pods for that operator)
3. Add to existing amount of pods to initial count 
4. Repeat for as many operators as we want to check for
5. do a final get of all pods in "openshift-*" namespace and make sure they match the number we calculated

Thoughts?

Reply:
Think pre action and post action might work. We can run the pre action script before the scenario which returns the pod count in the given namespaces ( openshift* for random pod killing scenario ) and then the post action script ( same script as pre action ) after the scenario. We can then compare the values to determine pass/fail without having to modify/specify the pod count check in the powerfulseal config provided there are no changes to the cluster like increase in the node count leading to more pods ( in case of daemonsets ).

The pre/post action scripts are supplied on the fly and can be changed before kicking off the kraken, so user can specify the namespace to monitor depending on what is specified in the powerfulseal config file.

We can also make this simple by asking user to set the expected pod count in the yaml by calculating it manually with an assumption that the env doesn't change during the run.